temporal-sdk/src/main/java/io/temporal/worker/tuning/SlotSupplier.java

```diff
@@ -36,16 +37,19 @@
 @Experimental
 public interface SlotSupplier<SI extends SlotInfo> {
   /**
-   * This function is called before polling for new tasks. Your implementation should block until a
-   * slot is available then return a permit to use that slot.
+   * This function is called before polling for new tasks. Your implementation should return a
+   * Promise that is completed with a {@link SlotPermit} when one becomes available.
+   *
+   * <p>These futures may be cancelled if the worker is shutting down or otherwise abandons the
+   * reservation. This can cause an {@link InterruptedException} to be thrown, in the thread running
+   * your implementation. You may want to catch it to perform any necessary cleanup, and then you
+   * should rethrow the exception.
```

Comment:
> and then you should rethrow the exception

They can't rethrow without wrapping because the checked exception was removed from this interface. We should add back `throws InterruptedException`, though could consider `throws Exception` as a way of saying we have code to handle any exception.

Reply:
Added Exception